### PR TITLE
Feature: Set layout on render

### LINKED ISSUE #276: Allow specifying the layout on render
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Currently, it seems as if a layout can only be specified when registering a template engine while it should be an option when rendering individual views.
Registering a property for each layout is a workaround but this duplicates a lot of code during runtime, allocating memory.

### Motivation

The sole purpose of layouts is that they can be chosen per view render. When I render the view, I know which layout to use.

### Example

```js
  fastify.register(pov, {
    engine: { ejs },
    root: config.viewdir,
    layoutRoot: config.layoutdir, // Optionally specify where to find all layouts
    includeViewExtension: true,
    defaultContext: {
      dev: process.env.NODE_ENV === 'development',
    },
  });
  
  // ...

  fastify.get('/', async (request, reply) => {
    reply.view('someview', {}, 'somelayout');
  });
```

### PR DESCRIPTION
**Pull request by danielkhan**
This adds the option to specify a property `layout` when rendering `ejs`, `dot`, `handlebar` and `eta` templates.
Providing a layout globally or on render is mutually exclusive and an error will be thrown if both are set.

Docs look as if they could use a bit more structure and will be proposed in a separate PR. 

Usage:
```js
return reply.view('home', {}, { layout: './layouts/default' });
```

closes #276

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Please remove the extra `devDependencies`, we use `standard` only.

**Comment by danielkhan** [comment]
> Please remove the extra `devDependencies`, we use `standard` only.

This was copied from fastify and introduced and approved by @mcollina in #277.
Just making sure before rebasing this branch.

File: index.js [hunk 1]
```diff
@@ -27,38 +27,48 @@ function fastifyView (fastify, opts, next) {
...
-    return
+  function layoutIsValid (_layoutFileName) {
+    if (type !== 'dot' && type !== 'handlebars' && type !== 'ejs' && type !== 'eta') {
+      throw new Error('Only Dot, Handlebars, EJS, and Eta support the "layout" option')
+    }
+
+    if (!hasAccessToLayoutFile(_layoutFileName, getDefaultExtension(type))) {
+      throw new Error(`unable to access template "${_layoutFileName}"`)
+    }
+    return 1
```
Comment (mcollina) [inline index.js:47]: why 1?

**Review by mcollina — APPROVED** [review]
lgtm with a question

File: index.js (hunk 1 above)
Comment (danielkhan) [reply index.js]: Good catch. The whole return statement unnecessary

**Comment by climba03003** [comment]
> > Please remove the extra `devDependencies`, we use `standard` only.
> 
> This was copied from fastify and introduced and approved by @mcollina in #277. Just making sure before rebasing this branch.

Does it means that the changes should be on #277 and it should not be in here?

**Comment by danielkhan** [comment]
> Does it means that the changes should be on #277 and it should not be in here?

This feature branch is a fork of the branch containing #277. That's why I can easily rebase this one if these changes are unwanted.

**Comment by danielkhan** [comment]
> Please remove the extra `devDependencies`, we use `standard` only.

This should now be all cleaned up

**Comment by climba03003** [comment]
> > Please remove the extra `devDependencies`, we use `standard` only.
> 
> This should now be all cleaned up

I think this PR can be closed because all your works is merged on #284
You can see that https://github.com/fastify/point-of-view/pull/278/files has no file changes

**Pull request closed without merging**